assets/js/base/context/providers/cart-checkout/payment-events/index.tsx

```diff
@@ -84,59 +82,50 @@ export const PaymentEventsProvider = ( {
 
 	const {
 		__internalSetPaymentProcessing,
-		__internalSetPaymentPristine,
+		__internalSetPaymentIdle,
 		__internalEmitPaymentProcessingEvent,
 	} = useDispatch( PAYMENT_STORE_KEY );
 
-	// flip payment to processing if checkout processing is complete, there are no errors, and payment status is started.
+	// flip payment to processing if checkout processing is complete, there are no errors
 	useEffect( () => {
 		if (
 			checkoutIsProcessing &&
 			! checkoutHasError &&
-			! checkoutIsCalculating &&
-			! isPaymentFinished
+			! checkoutIsCalculating
 		) {
 			__internalSetPaymentProcessing();
+
+			// Note: the nature of this event emitter is that it will bail on any
+			// observer that returns a response that !== true. However, this still
+			// allows for other observers that return true for continuing through
+			// to the next observer (or bailing if there's a problem).
+			__internalEmitPaymentProcessingEvent(
+				currentObservers.current,
+				setValidationErrors
+			);
 		}
 	}, [
 		checkoutIsProcessing,
 		checkoutHasError,
 		checkoutIsCalculating,
-		isPaymentFinished,
 		__internalSetPaymentProcessing,
+		__internalEmitPaymentProcessingEvent,
+		setValidationErrors,
 	] );
 
-	// When checkout is returned to idle, set payment status to pristine but only if payment status is already not finished.
+	// When checkout is returned to idle, and the payment setup has not completed, set payment status to idle
 	useEffect( () => {
-		if ( checkoutIsIdle && ! isPaymentSuccess ) {
-			__internalSetPaymentPristine();
+		if ( checkoutIsIdle && ! isPaymentReady ) {
+			__internalSetPaymentIdle();
 		}
-	}, [ checkoutIsIdle, isPaymentSuccess, __internalSetPaymentPristine ] );
+	}, [ checkoutIsIdle, isPaymentReady, __internalSetPaymentIdle ] );
```

Comment:
Just to check this logic; would the emitters etc run again after placing the order again? Because if the order changes (costs, selected payment method etc), payment setup may need to re-run. Maybe this is why it reset to pristine?

Reply:
Although I realise this is how it was before so may be a non-issue.

Reply:
I've just tested this by paying with a bad card (that declines payment), then changing the delivery rate, then pay with a successsful card. Works ok, so I don't think this is an issue. Anytime the Place Order button is clicked, the whole process restarts again (checkout status set to BEFORE_PROCESSING, then PROCESSING, then the payment status goes to PROCESSING, etc). There's also a useEffect that resets the payment status back to idle anytime the checkout is idle and there are no errors.